src/infrastructure/channels/reliable_channel.rs

```diff
@@ -141,9 +141,19 @@ impl Channel for ReliableChannel {
 
         self.dropped_packets = dropped_packets.into_iter().map(|(_, p)| p).collect();
 
-        let a = buffer.len();
-        let b = acked_header.size();
-        // TODO: resent packets if there are dropped packets
         Ok(&buffer[acked_header.size() as usize .. buffer.len()])
     }
+}
+
+impl ReliableChannel {
+    /// Check if this channel has dropped packets.
+    /// Performance wise it is better to use this to check if there are dropped packets than calling 'get_dropped_packets directly()'.
+    pub fn has_dropped_packets(&self) -> bool {
+        !self.dropped_packets.len() == 0
+    }
+
+    /// Creates a draining iterator that removes dropped packets and yield the ones that are removed.
+    pub fn drain_dropped_packets(&mut self) -> Vec<Box<[u8]>> {
+        return self.dropped_packets.drain(..).collect()
```

Comment:
could you explain why you use drain here?

Reply:
What is the specific question you are asking here? Is there a reason to not use drain?

Reply:
Yes, why is drain used here when you can just return the vector?

Reply:
@TimonPost The explanation for this is something that we should probably add to the docstring.

Reply:
I use drain because otherwise dropped packets will be queued up in the vector. Once you request dropped packets you are supposed to resend them if they will not be acknowledged thw packets will aromatically end in the dropped vector. If where were not to drain the collection we can't keep proper track of dropped packets.

I'll update the docstring